### ISSUE TITLE
tp-link KL-130 color not changing

### ISSUE THREAD

**Issue by ivandrofly** [CodeBardian/tplink-smartdevices-netstandard#6]
see: https://github.com/CodeBardian/tplink-smartdevices-netstandard/issues/5#issuecomment-686252075

**Comment by Jqnxyz** [comment]
Am also experiencing this issue on a LB-130. Any workaround?

**Comment by ivandrofly** [comment]
> Am also experiencing this issue on a LB-130. Any workaround?

I'm currently working on a patch or this.. maybe I need accelerate a bit :)

**Comment by Jqnxyz** [comment]
Saw your PR, I'd help but my .NET experience is extremely lacking hahaha

**Comment by ivandrofly** [comment]
> Saw your PR, I'd help but my .NET experience is extremely lacking hahaha

No worries, I got your back 😄 !

**Comment by CodeBardian** [comment]
I just checked on this issue and it seems like it is caused because the command to change colour is build wrong. Not because it does not allow multiple commands at once. There simply is not a **light_state** object, instead all arguments are directly passed with **transition_light_state**
So changing this: 
```csharp
Execute("smartlife.iot.smartbulb.lightingservice", "transition_light_state", "light_state", new JObject 
{
    new JProperty("hue", hsv.Hue),
    new JProperty("saturation", hsv.Saturation),
    new JProperty("brightness", (int)(hsv.Value * 100 / 255)),
});
```
to below, should solve the problem. At least it does for my KL-130, no clue about LB-130 yet :)
```csharp
Execute("smartlife.iot.smartbulb.lightingservice", "transition_light_state", new JObject
{
    new JProperty("hue", hsv.Hue),
    new JProperty("saturation", hsv.Saturation),
    new JProperty("brightness", (int)(hsv.Value * 100 / 255)),
}, null);
```

**Comment by ivandrofly** [comment]
> I just checked on this issue and it seems like it is caused because the command to change colour is build wrong. Not because it does not allow multiple commands at once. There simply is not a **light_state** object, instead all arguments are directly passed with **transition_light_state**
> So changing this:
> 
> ```cs
> Execute("smartlife.iot.smartbulb.lightingservice", "transition_light_state", "light_state", new JObject 
> {
>     new JProperty("hue", hsv.Hue),
>     new JProperty("saturation", hsv.Saturation),
>     new JProperty("brightness", (int)(hsv.Value * 100 / 255)),
> });
> ```
> 
> to below, should solve the problem. At least it does for my KL-130, no clue about LB-130 yet :)
> 
> ```cs
> Execute("smartlife.iot.smartbulb.lightingservice", "transition_light_state", new JObject
> {
>     new JProperty("hue", hsv.Hue),
>     new JProperty("saturation", hsv.Saturation),
>     new JProperty("brightness", (int)(hsv.Value * 100 / 255)),
> }, null);
> ```

Alright I will test your solution also and report back!
Thanks for information

**Comment by ivandrofly** [comment]
I haven't test your code yet, but I ` new JProperty("brightness", (int)(hsv.Value * 100 / 255)),`  will still 'cuase problem... the KL-130 model color are represented in degrees from 0-360

**Comment by CodeBardian** [comment]
> 
> 
> I haven't test your code yet, but ` new JProperty("brightness", (int)(hsv.Value * 100 / 255)),` will still 'cause problem... the KL-130 model color are represented in degrees from 0-360

Yes, that is true. I removed that nonsense in my tests too. when building the project from _color_changing_fix_ branch I am now able to apply every color I could think of.

**Comment by ivandrofly** [comment]
> Yes, that is true. I removed that nonsense in my tests too. when building the project from _color_changing_fix_ branch I am now able to apply every color I could think of.

Nice, if you have any uncommitted changes please push them so I can rebase my branch/PR onto that...
